File: 백준/Gold/1253. 좋다/좋다.py
from sys import stdin

N = int(stdin.readline().strip())
nums = list(map(int, stdin.readline().split()))
nums.sort()

answer = 0

for i in range(N):
    target = nums[i]
    left = 0
    right = N-1
    
    while left < right:
        if nums[left] + nums[right] == target:
            if left != i and right != i:
                answer += 1
                break
            elif left == i:
                left += 1
            else:
                right -= 1
            
        elif nums[left] + nums[right] < target:
            left += 1
        elif nums[left] + nums[right] > target:
            right -= 1

print(answer)

File: 백준/Gold/1253. 좋다/좋다.java
import java.io.*;
import java.util.*;

public class Main {
    static int answer = 0;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());
        int[] nums = new int[N];
        for (int i = 0; i < N; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }
        Arrays.sort(nums);

        for (int i = 0; i < N; i++) {
            int target = nums[i];
            int left = 0;
            int right = N - 1;

            while (left < right) {
                if (nums[left] + nums[right] == target) {
                    if (left != i && right != i) {
                        answer++;
                        break;
                    } else if (left == i) {
                        left++;
                    } else {
                        right--;
                    }
                } else if (nums[left] + nums[right] < target) {
                    left++;
                } else if (nums[left] + nums[right] > target) {
                    right--;
                }
            }
        }

        System.out.println(answer);
    }
}